core/src/executor/join.rs

```diff
@@ -273,10 +270,9 @@ impl<'a> JoinExecutor<'a> {
                     .await?
                     .try_into()?;
 
-                    let hash_key = match hash_key {
-                        Some(hash_key) => hash_key,
-                        None => return Ok(None),
-                    };
+                    if matches!(hash_key, Key::None) {
+                        return Ok(None);
+                    }
```

Comment:
Rather than using this as `Option<Key>`,
Is there any reason to add a variant named None to the key enum?

Reply:
Good question, I also actually tried `Option<Key>` first :)
Then, I changed it to just use `Key` with `Key::None`.
Though existing three cases handle the `None` case of the key slightly differently, those were the same that they all use `None`.
This meant that for all three cases I needed to specify `Option<Key>` and there were no use cases of using `Key` without `Option`, so I changed the codes to include `None` in the `Key`.

`Key::None` is actually closer to `Null`. However in this case `Key::None == Key::None` but `Value::Null != Value::Null`, so I thought that using `Null` keyword is not a good idea.